pkgs/development/libraries/science/math/faiss/pytest-suite.nix

```diff
@@ -0,0 +1,30 @@
+{ lib
```

Comment:
Why is this file necessary? Can't the tests be handled in the main derivation?

Reply:
I could write a custom `checkPhase ` instead, but since the linkage takes rather long for this package, I tried to untangle the derivation and the tests. I grepped for modules that potentially could do something similar first and there were e.g. matches for running `pytest` in `passthru.tests`.

On a side note, I think separating long tests from long builds is desirable for big packages, including some that are already in nixpkgs. I cannot defend the particular structure though

Reply:
ok that makes sense to me... just was confusing at first glance!

Reply:
I should be happy to rewrite this so as to reduce confusion! If only you got an idea

Reply:
I'm also used this approach and put tests into a seperate file `tests.nix`to avoid infinite recursion, you can have a look here  https://github.com/NixOS/nixpkgs/tree/80ec2d1ace7c133ce640a3ebc72130b6c211039f/pkgs/development/python-modules/doit